src/test/java/test/InvokedMethodNameListener.java

```diff
@@ -141,7 +141,7 @@ private static String getName(ITestResult result) {
   }
 
   public List<String> getSucceedMethodNames() {
-    return Collections.unmodifiableList(succeedMethodNames);
+    return new ArrayList<>(succeedMethodNames);
```

Comment:
Come to think of it, it would be nice to still return an unmodifiable list that copies the original one (unlike `unmodifiableList`, which merely wraps it). Not sure this exists, though.